src/lib/evil/evil_stat.c

```diff
@@ -0,0 +1,126 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "evil_private.h"
+#include <evil_api.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include "evil_stat.h"
+
+#define	AT_SYMLINK_NOFOLLOW	0x01
+
+EAPI int 
+fstatat(int dirfd, const char *pathname, struct stat *statbuf, int flags)
+{
+   int r_fstatat;
+
+   if (pathname[2] == '\\')
+     {  
+        if (flags == AT_SYMLINK_NOFOLLOW)
+          r_fstatat = stat(pathname, statbuf);
+        else
+          r_fstatat = stat(pathname, statbuf);
+        return r_fstatat;
+     }
+   else
+     {
+        char *pathbuf = NULL;
+        size_t pathbuf_size = 0;
+        DWORD copied = 0;
+        do
+        {
+           pathbuf_size += MAX_PATH;
+           pathbuf = realloc(pathbuf, pathbuf_size * sizeof(char));
+           copied = GetModuleFileName(NULL, pathbuf, pathbuf_size);
+        } while (copied >= pathbuf_size);
+       
+        pathbuf_size = copied;
+        pathbuf = realloc(pathbuf, pathbuf_size * sizeof(char)); 
+        copied = GetModuleFileName(NULL, pathbuf, pathbuf_size);
+
+        int size_str;
+        for (size_str = strlen(pathbuf); size_str >= 0; size_str -- )
+         {
+            if (pathbuf[size_str] == '\\')
+              {
+                 pathbuf[size_str + 1] = 0;
+                 break;
+              }
+         }
```

Comment:
What if the path separator is `/`? Furthermore, this loop can be replace by [_splitpath](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/splitpath-wsplitpath?redirectedfrom=MSDN&view=vs-2019)

Reply:
so, I based myself on Windows, to put the "/" separator, as for the loop, I will study: _splitpath.